src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.MainApp;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.logic.CommandHistory;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.util.SampleDataUtil;
+import seedu.address.storage.AddressBookStorage;
+import seedu.address.storage.JsonAddressBookStorage;
+import seedu.address.storage.StorageManager;
+
+/**
+ * Imports records from a text file.
+ */
+public class ImportCommand extends Command {
+
+    public static final String COMMAND_WORD = "import";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Imports a text file in the \"data\" folder \n"
+            + "Parameters: FILENAME\n"
+            + "Example: " + COMMAND_WORD + " records1.json";
+
+    public static final String MESSAGE_SUCCESS = "Imported the records!";
+
+    private final File file;
+
+    public ImportCommand(File file) {
+        this.file = file;
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) {
+        requireNonNull(model);
+        readFile(model);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    /**
+     * readFile() overwrites the current address book with the contents of the file.
+     */
+    private void readFile(Model model) {
+        AddressBookStorage addressBookStorage = new JsonAddressBookStorage(file.toPath());
+
+        StorageManager storage = new StorageManager(addressBookStorage, null);
+
+        final Logger logger = LogsCenter.getLogger(MainApp.class);
+
+        Optional<ReadOnlyAddressBook> addressBookOptional;
+        ReadOnlyAddressBook initialData;
+
+        try {
+            addressBookOptional = storage.readAddressBook();
+            if (!addressBookOptional.isPresent()) {
+                logger.info("Data file not found. Will be starting with a sample AddressBook");
+            }
+            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+        } catch (DataConversionException e) {
```

Comment:
Good that you are handling the exception cases without crashing/exiting the application